src/mgmt/srv_query.c

```diff
@@ -319,6 +319,17 @@ ds_mgmt_smd_list_devs(Mgmt__SmdDevResp *resp)
 		strncpy(resp->devices[i]->state,
 			bio_dev_state_enum_to_str(state), buflen);
 
+		D_ALLOC(resp->devices[i]->traddr, buflen);
+		if (resp->devices[i]->traddr == NULL) {
+			D_ERROR("Failed to allocate device transport ID");
+			rc = -DER_NOMEM;
+			break;
+		}
+
+		/* Transport Addr -> Blobstore UUID mapping */
+		strncpy(resp->devices[i]->traddr, dev_info->bdi_traddr,
+			buflen);
```

Comment:
The bdi_traddr could be NULL (for AIO devices, or failed to get traddr for whatever reason), so we need check bdi_traddr != NULL first, then allocate traddr for resp with length = strlen(bdi_traddr) + 1, then strncpy.

Reply:
updated